.github/workflows/qa.yaml

```diff
@@ -9,14 +9,48 @@ on:
       - "renovate/*"
   pull_request:
 
+# Cancel existing jobs for this branch/PR
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
 jobs:
   lint:
     uses: canonical/starflow/.github/workflows/lint-python.yaml@main
   test:
-    uses: canonical/starflow/.github/workflows/test-python.yaml@main
+    uses: canonical/starflow/.github/workflows/test-python.yaml@work/setup-vars
     with:
       fast-test-platforms: '["ubuntu-22.04", "ubuntu-24.04"]'
       fast-test-python-versions: '["3.10", "3.12", "3.13"]'
       slow-test-platforms: '["ubuntu-24.04"]'
       slow-test-python-versions: '["3.12"]'
       lowest-python-version: ""
+      pytest-markers: not java and not python and not plugin
+      setup-vars: NO_JAVA=1 NO_PYTHON=1 NO_PLUGIN=1
+  test-java-plugins:
+    uses: canonical/starflow/.github/workflows/test-python.yaml@work/setup-vars
+    with:
+      fast-test-platforms: "[]"
+      slow-test-platforms: '["ubuntu-24.04"]'
+      slow-test-python-versions: '["3.12"]'
+      lowest-python-version: ""
+      pytest-markers: java
+      setup-vars: NO_PYTHON=1 NO_PLUGIN=1
+  test-python-plugins:
+    uses: canonical/starflow/.github/workflows/test-python.yaml@work/setup-vars
+    with:
+      fast-test-platforms: "[]"
+      slow-test-platforms: '["ubuntu-24.04"]'
+      slow-test-python-versions: '["3.12"]'
+      lowest-python-version: ""
+      pytest-markers: python
+      setup-vars: NO_JAVA=1 NO_PLUGIN=1
```

Comment:
thought: I wonder how hard that will be to maintain without getting it wrong at some point, especially if we want to split another group of test out of the common group. We will need to change the setup-vars of every job. However I do not see a better solution :/.

Reply:
I made it opt-out to preserve the UX for the more important use case of running setup on one's own machine. If we forget an opt out when changing CI, the only real downside is the installation of additional unnecessary packages.